### PR TITLE
GLOB-68575: Update CI config and add description field to graph object metadata

### PR DESCRIPTION
**Pull request by GaryGlobality**
This change has two purposes:
1. Update the CI config using `globality-build circle-gen local && globality-build build-gen local` to fix the CircleCI builds.
2. Add a `description` field to the `Metadata` class to support describing objects.

## Why?

- To allow future PRs to have successful builds.
- To support improving the API documentation generation by allowing graph objects to be described.

### CODE REVIEW

File: microcosm/tests/test_object_graph.py [hunk 1]
```diff
@@ -163,7 +162,20 @@ def test_get_component_name_function():
...
+
+
+def test_describe_object_graph():
+    graph = create_object_graph(
+        name="test", description="A description of the object graph."
+    )
     assert_that(
-        get_component_name(graph, graph.parent),
-        is_(equal_to("parent"))
+        graph.metadata.description, is_(equal_to("A description of the object graph."))
```
Comment (ronanm-globality) [inline microcosm/tests/test_object_graph.py:173]: Sounds good. Do you have a specific case in mind to use this in for context? 🤔

**Review by ronanm-globality — APPROVED** [review]

File: microcosm/tests/test_object_graph.py (hunk 1 above)
Comment (GaryGlobality) [reply microcosm/tests/test_object_graph.py:173]: The description will be set on the `sphinx` graph object and used to generate a top-level description of its API in the auto-generated swagger docs.

File: Dockerfile [hunk 2]
```diff
@@ -12,8 +12,7 @@
 #
 
 # ----------- deps -----------
-# Install from Debian Stretch with modern Python support
-FROM python:slim-stretch as deps
+FROM python:3.7-stretch as deps
```
Comment (jpringle-globality) [inline Dockerfile:15]: This caught my eye. Are any services running Python 3.7 and using Microcosm? This version should be the lowest version of our Python services that use Microcosm, right?

**Review by jpringle-globality — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (GaryGlobality) [reply Dockerfile:15]: Yes, `sphinx` for [example](https://github.com/globality-corp/sphinx/blob/master/Dockerfile#L17). AFAIK none of them pin to specific patch version - I found this in our [Python wiki page](https://globality.atlassian.net/wiki/spaces/DEV/pages/927861472/Python#Standard/X86-Installation):

> We don’t pin to a specific bugfix release – you can install the latest.

**Pull request merged**